### PR TITLE
feat: add voice-related functions & getters

### PR DESCRIPTION
**Pull request by MilSci**
**Please describe the changes this PR makes and why it should be merged:**
- get VoiceStates() function was added to get the voice states of a channel
- get connectedMembers() function was added to get the connected voice members of a channel
- disconnectMember(memberID) function was added to remove a member from the voice channel

**Status**

- [x] Code changes have been tested against the Discord API, or there are no
  code changes

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed,
    parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation,
  README, etc.

### CODE REVIEW

File: src/api/structures/channel.ts [hunk 1]
```diff
@@ -22,6 +24,36 @@ const baseChannel: Partial<Channel> = {
...
+      .map((vs) => {
+        return cache.members.get(vs.member!.user.id);
+      });
+    const memberCollection: Collection<
+      string,
+      Member | undefined
+    > = new Collection(members!.map((m) => [m!.id, m!]));
+    return memberCollection;
+  },
+  disconnectMember(memberID) {
```
Comment (ayntee) [inline src/api/structures/channel.ts:52]:
```suggestion
  disconnectMember() {
```

File: src/api/structures/channel.ts [hunk 2]
```diff
@@ -22,6 +24,36 @@ const baseChannel: Partial<Channel> = {
...
+    const channel = cache.channels.get(this.id!);
+    const guildID = channel!.guild!.id;
+    kickFromVoiceChannel(guildID, memberID);
```
Comment (ayntee) [inline src/api/structures/channel.ts:55]:
```suggestion
    kickFromVoiceChannel(this.guildID, memberID);
```

File: src/api/structures/channel.ts [hunk 3]
```diff
@@ -22,6 +24,36 @@ const baseChannel: Partial<Channel> = {
   get mention() {
     return `<#${this.id!}>`;
   },
+  get voiceStates() {
+    const channel = cache.channels.get(this.id!);
+    const voiceStates = channel!.guild?.voiceStates.filter(
+      (vs) => vs.channelID === channel?.id
+    );
+    return voiceStates;
```
Comment (ayntee) [inline src/api/structures/channel.ts:32]:
```suggestion

    return voiceStates;
```

File: src/api/structures/channel.ts (hunk 2 above)
Comment (itohatweb) [inline src/api/structures/channel.ts:55]:
```suggestion
    kickFromVoiceChannel(this.guildID!, memberID);
```

**Review by itohatweb — CHANGES_REQUESTED** [review]

File: src/api/structures/channel.ts [hunk 4]
```diff
@@ -22,6 +24,36 @@ const baseChannel: Partial<Channel> = {
   get mention() {
     return `<#${this.id!}>`;
   },
+  get voiceStates() {
+    const channel = cache.channels.get(this.id!);
+    const voiceStates = channel!.guild?.voiceStates.filter(
+      (vs) => vs.channelID === channel?.id
+    );
```
Comment (ayntee) [inline src/api/structures/channel.ts:31]:
```suggestion
    const guild = cache.guilds.get(this.guildID);
    if (!guild) return;
    return guild.voiceStates.filter(voiceState => voiceState.channelID === this.id);
```

File: src/api/structures/channel.ts [hunk 5]
```diff
@@ -22,6 +24,34 @@ const baseChannel: Partial<Channel> = {
   get mention() {
     return `<#${this.id!}>`;
   },
+  get voiceStates() {
+    const guild = cache.guilds.get(this.guildID);
+    if (!guild) return;
+    return guild.voiceStates.filter(voiceState => voiceState.channelID === this.id);
+
+    return voiceStates;
```
Comment (ayntee) [inline src/api/structures/channel.ts:32]:
```suggestion
```

File: src/api/structures/channel.ts [hunk 6]
```diff
@@ -22,6 +24,32 @@ const baseChannel: Partial<Channel> = {
   get mention() {
     return `<#${this.id!}>`;
   },
+  get voiceStates() {
+    const guild = cache.guilds.get(this.guildID);
+    if (!guild) return;
```
Comment (MilSci) [inline src/api/structures/channel.ts:29]: if(!guild) return undefined;

File: src/api/structures/channel.ts [hunk 7]
```diff
@@ -22,6 +24,32 @@ const baseChannel: Partial<Channel> = {
...
+        );
+      })
+      .map((vs) => {
+        return cache.members.get(vs.member!.user.id);
+      });
+    const memberCollection: Collection<
+      string,
+      Member | undefined
+    > = new Collection(members!.map((m) => [m!.id, m!]));
+    return memberCollection;
```
Comment (itohatweb) [inline src/api/structures/channel.ts:48]:
```suggestion
  get connectedMembers() {
    return new Collection(
      this.voiceStates?.map(
        (vs) => [vs.member?.user.id!, cache.members.get(vs.member?.user.id!)]
      ) || [],
    );
  },
```

File: src/api/structures/channel.ts [hunk 8]
```diff
@@ -22,6 +24,22 @@ const baseChannel: Partial<Channel> = {
   get mention() {
     return `<#${this.id!}>`;
   },
+  get voiceStates() {
+    const guild = cache.guilds.get(this.guildID);
+    if (!guild) return;
```
Comment (itohatweb) [inline src/api/structures/channel.ts:29]:
```suggestion
    const guild = cache.guilds.get(this.guildID!);
    if (!guild) return undefined;
```

File: src/api/structures/channel.ts [hunk 9]
```diff
@@ -22,6 +24,22 @@ const baseChannel: Partial<Channel> = {
...
+    return guild.voiceStates.filter(voiceState => voiceState.channelID === this.id);
+  },
+  get connectedMembers() {
+    return new Collection(
+      this.voiceStates?.map(
+        (vs) => [vs.member?.user.id!, cache.members.get(vs.member?.user.id!)]
+      ) || [],
+    );
+  },
+  },
```
Comment (itohatweb) [inline src/api/structures/channel.ts:39]:
```suggestion
```

File: src/api/structures/channel.ts [hunk 10]
```diff
@@ -22,6 +24,22 @@ const baseChannel: Partial<Channel> = {
...
+  disconnectMember(memberID) {
+    kickFromVoiceChannel(this.guildID, memberID);
```
Comment (itohatweb) [inline src/api/structures/channel.ts:41]:
```suggestion
    kickFromVoiceChannel(this.guildID!, memberID);
```

**Review by itohatweb — CHANGES_REQUESTED** [review]

File: src/api/structures/channel.ts [hunk 11]
```diff
@@ -22,6 +24,21 @@ const baseChannel: Partial<Channel> = {
...
+    return guild.voiceStates.filter(voiceState => voiceState.channelID === this.id);
+  },
+  get connectedMembers() {
+    return new Collection(
+      this.voiceStates?.map(
+        (vs) => [vs.member?.user.id!, cache.members.get(vs.member?.user.id!)]
+      ) || [],
+    );
+  },
+  disconnectMember(memberID) {
```
Comment (itohatweb) [inline src/api/structures/channel.ts:39]: Please rename to `kickFromVoice` for consistency.

File: src/api/structures/channel.ts [hunk 12]
```diff
@@ -22,6 +24,21 @@ const baseChannel: Partial<Channel> = {
   get mention() {
     return `<#${this.id!}>`;
   },
+  get voiceStates() {
+    const guild = cache.guilds.get(this.guildID!);
+    if (!guild) return undefined;
+    return guild.voiceStates.filter(voiceState => voiceState.channelID === this.id);
```
Comment (Skillz4Killz) [inline src/api/structures/channel.ts:30]: Always a space after `return` add empty line plz

File: src/api/structures/channel.ts [hunk 13]
```diff
@@ -22,6 +24,21 @@ const baseChannel: Partial<Channel> = {
   get mention() {
     return `<#${this.id!}>`;
   },
+  get voiceStates() {
+    const guild = cache.guilds.get(this.guildID!);
+    if (!guild) return undefined;
```
Comment (Skillz4Killz) [inline src/api/structures/channel.ts:29]:
```suggestion
    if (!guild) return new Collection();
```

File: src/api/structures/channel.ts [hunk 14]
```diff
@@ -22,6 +24,21 @@ const baseChannel: Partial<Channel> = {
...
+  },
+  get connectedMembers() {
+    return new Collection(
+      this.voiceStates?.map(
+        (vs) => [vs.member?.user.id!, cache.members.get(vs.member?.user.id!)]
+      ) || [],
+    );
+  },
+  kickFromVoice(memberID) {
+    kickFromVoiceChannel(this.guildID!, memberID);
```
Comment (Skillz4Killz) [inline src/api/structures/channel.ts:40]:
```suggestion
    return kickFromVoiceChannel(this.guildID!, memberID);
```

File: src/api/structures/channel.ts [hunk 15]
```diff
@@ -112,9 +129,16 @@ export interface Channel {
...
+  /** The connected voice members */
+  connectedMembers: Collection<string, Member | undefined>;
+  /** The connected voice members */
+  voiceStates: Collection<string, CleanVoiceState> | undefined;
 
   // METHODS
 
+  /** Disconnect a member from a voice channel. Requires MOVE_MEMBERS permission. */
+  kickFromVoice(memberID: string): void;
+
```
Comment (Skillz4Killz) [inline src/api/structures/channel.ts:141]: remove extra line

File: src/api/structures/channel.ts [hunk 16]
```diff
@@ -112,9 +129,16 @@ export interface Channel {
...
   mention: string;
+  /** The connected voice members */
+  connectedMembers: Collection<string, Member | undefined>;
+  /** The connected voice members */
+  voiceStates: Collection<string, CleanVoiceState> | undefined;
 
   // METHODS
 
+  /** Disconnect a member from a voice channel. Requires MOVE_MEMBERS permission. */
+  kickFromVoice(memberID: string): void;
```
Comment (Skillz4Killz) [inline src/api/structures/channel.ts:140]:
```suggestion
  kickFromVoice(memberID: string): ReturnType<typeof kickFromVoiceChannel >;
```

File: src/api/structures/guild.ts [hunk 17]
```diff
@@ -345,12 +349,14 @@ export interface Guild {
   unban(memberID: string): ReturnType<typeof unban>;
   /** Get all the invites for this guild. Requires MANAGE_GUILD permission */
   invites(): ReturnType<typeof getInvites>;
+  /** Kicks the user from a voice channel. Requires MOVE_MEMBERS permission*/
+  kickFromVoice(memberID: string): void;
```
Comment (Skillz4Killz) [inline src/api/structures/guild.ts:353]:
```suggestion
  kickFromVoice(memberID: string): ReturnType<typeof kickFromVoiceChannel>;
```

File: src/api/structures/member.ts [hunk 18]
```diff
@@ -178,6 +182,8 @@ export interface Member {
   sendDM(
     content: string | MessageContent,
   ): ReturnType<typeof sendDirectMessage>;
+  /** Kick the member from a voice channel */
+  kickFromVoice(guildID: string): void;
```
Comment (Skillz4Killz) [inline src/api/structures/member.ts:186]:
```suggestion
  kickFromVoice(guildID: string): ReturnType<typeof kickFromVoiceChannel>;
```

**Review by Skillz4Killz — CHANGES_REQUESTED** [review]

File: src/api/structures/channel.ts [hunk 19]
```diff
@@ -22,6 +24,21 @@ const baseChannel: Partial<Channel> = {
   get mention() {
     return `<#${this.id!}>`;
   },
+  get voiceStates() {
+    const guild = cache.guilds.get(this.guildID!);
+    if (!guild) return undefined;
+    return guild.voiceStates.filter(voiceState => voiceState.channelID === this.id);
+  },
+  get connectedMembers() {
+    return new Collection(
```
Comment (Skillz4Killz) [inline src/api/structures/channel.ts:33]: This thing might be better as ```ts
return this.voiceStates.map((vs, key) => [key, cache.members.get(key)])
```

**Review by Skillz4Killz — CHANGES_REQUESTED** [review]
Still pending changes from previous reviews

**Pull request closed without merging**